src/components/validation/resources/http.rs

```diff
@@ -235,8 +240,33 @@ fn spawn_input_http_client(
             debug!("Got event to send from runner.");
 
             let mut buffer = BytesMut::new();
+            if let (
+                Json(_),
+                Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }),
+            ) = (encoder.serializer(), encoder.framer())
+            {
+                buffer.put_u8(b'[');
+            }
```

Comment:
It might be worth caching this decision since you use it below:
```suggestion
            let is_json = matches!(encoder.serializer(), Json(_)) && matches!(encoder.framer(), Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }));
            if is_json {
                buffer.put_u8(b'[');
            }
```